packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -89,8 +94,24 @@ export class SerializedStateManager {
 		return this._offlineLoadEnabled;
 	}
 
+	private refreshOnCatchUpMaybe(message: ISequencedDocumentMessage) {
```

Comment:
This approach is taking into account that we're only going to refresh the base snapshot once at loading. If we do it more frequently, we could get rid of all new latest variables and call refreshSnapshot() from addProcessedOp instead.